File: samples/javascript/notepad.js
var selenium = require('selenium-webdriver');
var reporters = require('jasmine-reporters');

var until = selenium.until;

var junitReporter = new reporters.JUnitXmlReporter({
  savePath: __dirname,
  consolidateAll: true
});
jasmine.getEnv().addReporter(junitReporter);

jasmine.DEFAULT_TIMEOUT_INTERVAL = 30000

describe('InsuranceWeb Tests', function () {
  beforeAll(function (done) {

    jasmine.DEFAULT_TIMEOUT_INTERVAL = 30000

    new selenium.Builder()
      .usingServer('http://localhost:8080')
      .withCapabilities({"app" : "%WINDIR%\\notepad.exe"})
      .forBrowser("firefox")
      .build()
      .then(function (innerDriver) {
        driver = innerDriver;

        driver.executeScript("appdriver:startTrueLog", process.cwd() + "\\result.tlz").then(done);
      });
  });

  beforeEach(function (done) {
    done();
  });

  afterEach(function (done) {
    done();
  });

  it('should type something', function (done) {
    driver.findElement(selenium.By.xpath('//TextField')).sendKeys("Hello from JavaScript!")
    .then(_ => driver.findElement(selenium.By.xpath("//Menu[@caption='File']")).click())
    .then(_ => driver.findElement(selenium.By.xpath("//MenuItem[@caption='Exit']")).click())
    .then(_ => driver.findElement(selenium.By.xpath("//PushButton[@caption=\"Don't Save\"]")).click())

    .then(done);
  });

  afterAll(function (done) {
    driver.executeScript("appdriver:stopTrueLog")
    .then(_ => driver.quit())
    .then(done);
  });
});

File: samples/python/notepad.py
import os
import unittest
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.select import Select
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

class InsuranceTests(unittest.TestCase):
    @classmethod
    def setUpClass(cls):
        cls.driver = driver = webdriver.Remote(
            command_executor='http://127.0.0.1:8080',
            desired_capabilities={'app': '%WINDIR%\\notepad.exe'})

        cls.wait = WebDriverWait(cls.driver, 10)

    def setUp(self):        
        self.driver.execute_script("appdriver:startTrueLog",
            os.path.dirname(os.path.abspath(__file__)) + "\\results\\" + __class__.__name__ + ".tlz")

    def test1(self):            
        self.assertEqual(self.driver.title, "Untitled - Notepad")
        self.driver.set_window_rect(400, 500, 300, 200)

        window = self.driver.find_element_by_xpath("//Window")

        textfield = window.find_element_by_xpath("//TextField")
        textfield.clear()
        textfield.send_keys("hello from python!")
        
        self.assertEqual(textfield.text, "hello from python!")

        filemenu = self.driver.find_element_by_xpath("//Menu[@caption='File']")
        filemenu.click()

        self.driver.find_element_by_xpath("//MenuItem[@caption='Exit']").click()

        self.driver.find_element_by_xpath("//PushButton[@caption=\"Don't Save\"]").click()

    @classmethod
    def tearDownClass(cls):
        cls.driver.execute_script("appdriver:stopTrueLog")
        cls.driver.quit()

if __name__ == "__main__":
    unittest.main()
